advanced-cache.php

```diff
@@ -566,20 +566,21 @@ function add_debug_html_to_output( $debug_html ) {
 } else if ( $batcache->seconds < 1 || $batcache->times < 1 ) {
 	// Are we only caching frequently-requested pages?
 	$batcache->do = true;
-} else {
-	// No batcache item found, or ready to sample traffic again at the end of the batcache life?
-	if ( !is_array($batcache->cache) || time() >= $batcache->cache['time'] + $batcache->max_age - $batcache->seconds ) {
-		wp_cache_add($batcache->req_key, 0, $batcache->group);
-		$batcache->requests = wp_cache_incr($batcache->req_key, 1, $batcache->group);
-
-		if ( $batcache->requests >= $batcache->times &&
-			time() >= $batcache->cache['time'] + $batcache->cache['max_age']
-		) {
-			wp_cache_delete( $batcache->req_key, $batcache->group );
-			$batcache->do = true;
-		} else {
-			$batcache->do = false;
-		}
+} else if ( ! is_array( $batcache->cache ) ) {
+	// No batcache item found, create it.
+	$batcache->do = true;
```

Comment:
I think this needs to do the cache add and cache incr too?

Reply:
Are they not just added at the end of page generation? I don't really know the inner workings of this too intimately.

Reply:
Not certain, just it's behaviour that was there before, so we should replicate imo.

Reply:
Oh, I am wrong! It'll be cleared by the `wp_cache_delete` that was there before.

I think all we need to do then is `$batcache->requests = 1`, not the cache entries

Reply:
Fixed!